src/routing_node.rs

```diff
@@ -939,6 +963,13 @@ impl RoutingNode {
         }
     }
 
+    fn connect(&mut self, endpoints: &Vec<::crust::Endpoint>) {
+        for e in endpoints {
+            self.connect_requests.insert(e.clone());
```

Comment:
this is not okay :).  We do not want to keep any state on ::connect; but I think I can easily enough fix it, so I'll raise an issue and take it out afterwards